sys/kern/subr_intr.c

```diff
@@ -876,8 +900,8 @@ intr_pic_deregister(device_t dev, intptr_t xref)
  *     an interrupts property and thus no explicit interrupt parent."
  */
 int
-intr_pic_claim_root(device_t dev, intptr_t xref, intr_irq_filter_t *filter,
-    void *arg)
+intr_pic_claim_root_num(device_t dev, intptr_t xref, intr_irq_filter_t *filter,
+    void *arg, uint32_t root)
```

Comment:
Perhaps there should be a reserved value for `root` which specifies a particular root controller is dynamic and needs `intr_pic_claim_root_num()` to return its index?